app/src/main/kotlin/com/intive/picover/profile/view/ProfileScreen.kt

```diff
@@ -9,78 +9,105 @@ import androidx.compose.material3.Divider
 import androidx.compose.material3.LinearProgressIndicator
 import androidx.compose.runtime.Composable
 import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.runtime.collectAsState
 import androidx.compose.runtime.getValue
 import androidx.compose.ui.Modifier
 import androidx.compose.ui.res.stringResource
 import androidx.compose.ui.unit.dp
-import androidx.navigation.NavHostController
+import cafe.adriel.voyager.core.screen.Screen
+import cafe.adriel.voyager.hilt.getScreenModel
 import com.intive.picover.R
 import com.intive.picover.common.error.PicoverGenericError
 import com.intive.picover.common.result.TakePictureOrPickImageContract
 import com.intive.picover.common.result.launch
-import com.intive.picover.common.viewmodel.state.ViewModelState.Error
-import com.intive.picover.common.viewmodel.state.ViewModelState.Loaded
-import com.intive.picover.common.viewmodel.state.ViewModelState.Loading
+import com.intive.picover.common.viewmodel.state.MVIStateType.ERROR
+import com.intive.picover.common.viewmodel.state.MVIStateType.LOADED
+import com.intive.picover.common.viewmodel.state.MVIStateType.LOADING
 import com.intive.picover.main.navigation.NavControllerHolder
 import com.intive.picover.profile.model.Profile
+import com.intive.picover.profile.model.ProfileState
 import com.intive.picover.profile.model.ProfileUpdateResult
 import com.intive.picover.profile.viewmodel.ProfileViewModel
 import dev.gitlive.firebase.storage.File
 
+class ProfileScreen : Screen {
+
+	@Composable
+	override fun Content() {
+		val viewModel = getScreenModel<ProfileViewModel>()
+		val state by viewModel.state.collectAsState()
+		val takePictureOrPickImageLauncher = rememberLauncherForActivityResult(TakePictureOrPickImageContract()) { uri ->
+			uri?.let { viewModel.updateAvatar(File(it)) }
+		}
+		LaunchedEffect(Unit) {
+			NavControllerHolder.observeResult<ProfileUpdateResult>().collect { viewModel.onProfileUpdateResult(it) }
+		}
+		ProfileContent(
+			state = state,
+			onEditPhotoClick = takePictureOrPickImageLauncher::launch,
+			onEditNameClick = { NavControllerHolder.navigate("updateProfile/${state.profile!!.name}") },
+			onLogoutClick = viewModel::onLogoutClick,
+			onDeleteAccountCLick = viewModel::onDeleteAccountClick,
+			onRetryClick = viewModel::fetchProfile,
+		)
+		if (state.showDeleteAccountPopup) {
```

Comment:
Voyager doesn't support dialog destinations

Reply:
So in the future we can replace them just by bottom sheets

Reply:
We could, but on the other hand it's sometimes useful to have some kind of popup 😄 I would keep it to have a point of reference how to deal with dialogs, maybe they will introduce some solution for that in voyager in the future